// @bfme-technology/styled/index.js
export {
  documentationBaseStyles,
  documentationBaseLayoutStyles,
} from "./src/documentationBaseStyles.js";

export {
  layoutWrapperClass,
  getSidebarClass,
  logoAdminClass,
  navClass,
  getLayoutNavLinkClass,
  navIconClass,
  navLinkTextClass,
  layoutMainClass,
  topbarClass,
  topbarTitleClass,
  themeToggleContainerClass,
  getThemeToggleButtonClass,
  contentAreaClass,
  breadcrumbContainerClass,
  breadcrumbListClass,
  breadcrumbItemClass,
  breadcrumbLinkClass,
  breadcrumbActiveItemClass,
  breadcrumbSeparatorClass,
  dashboardContainerClass,
  dashboardRootClass,
  dashboardTitleClass,
  metricsGridClass,
  metricCardClass,
  metricCardTitleClass,
  metricValueClass,
  gridContainerClass,
  gridPageWrapperClass,
  gridHeaderSectionClass,
  gridHeaderTitleClass,
  gridActionsSectionClass,
  gridFilterSectionClass,
  gridFilterLabelClass,
  gridFilterSelectClass,
  gridPrimaryButtonClass,
  gridContentSectionClass,
  gridEmptyTextClass,
} from "./src/studioAdminStyles.js";

import * as freyjaStyles from "./src/studioAdminStyles.js";
import * as mercuryStyles from "./src/themes/mercury.js";

export const freyja = freyjaStyles;
export const mercury = mercuryStyles;
